spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/Schedule.java

```diff
@@ -65,4 +66,18 @@
      * @return true if the job executes on time, otherwise false.
      */
     Boolean runningOnTime(Instant lastExecutionTime);
+
+    /**
+     * Gets the delay parameter of the schedule.
+     *
+     * @return the delay parameter of the schedule as a Long.
+     */
+    Long getDelay();
+
+    /**
+     * Sets the delay parameter of the schedule.
+     *
+     * @param delay the delay parameter.
+     */
+    void setDelay(Long delay);
```

Comment:
do we want to allow plugins to dynamically change the delay after initialization? that may cause the schedule harder to track

Reply:
I am curious, do we need to track the schedule of individual jobs for each of the plugins? I can foresee some contention is cases where one plugin needs to schedule some jobs and another plugin which depends on the same job.